cloudinit/config/cc_lxd.py

```diff
@@ -281,16 +281,25 @@ def maybe_cleanup_default(net_name, did_init, create, attach,
         return
 
     fail_assume_enoent = "failed. Assuming it did not exist."
+    failed = "failed."
     succeeded = "succeeded."
     if create:
-        msg = "Deletion of lxd network '%s' %s"
+        msg = "Detach of lxd network '%s' from profile '%s' %s"
         try:
-            _lxc(["network", "delete", net_name])
-            LOG.debug(msg, net_name, succeeded)
+            _lxc(["network", "detach-profile", net_name, profile])
+            LOG.debug(msg, net_name, profile, succeeded)
         except subp.ProcessExecutionError as e:
             if e.exit_code != 1:
                 raise e
-            LOG.debug(msg, net_name, fail_assume_enoent)
+            LOG.debug(msg, net_name, profile, fail_assume_enoent)
+        else:
+            msg = "Deletion of lxd network '%s' %s"
+            try:
+                _lxc(["network", "delete", net_name])
+                LOG.debug(msg, net_name, succeeded)
+            except subp.ProcessExecutionError as e:
+                LOG.debug(msg, net_name, failed)
```

Comment:
there is no reason to log this. it failed, we are raising the exception tot he caller.